app/src/main/java/com/infomaniak/mail/utils/extensions/Extensions.kt

```diff
@@ -493,38 +491,57 @@ inline val AndroidViewModel.context: Context get() = getApplication()
 val TextInputEditText.trimmedText inline get() = text?.trim().toString()
 
 fun Context.postfixWithTag(
-    original: Spannable,
+    original: CharSequence,
     @StringRes tagRes: Int,
-    @ColorRes backgroundColorRes: Int,
-    @ColorRes textColorRes: Int,
+    tagColor: TagColor,
+    ellipsizeConfiguration: SubjectFormatter.EllipsizeConfiguration? = null,
     onClicked: (() -> Unit)? = null,
-) = postfixWithTag(original, getString(tagRes), backgroundColorRes, textColorRes, onClicked)
-
-fun Context.postfixWithTag(
-    original: Spannable,
-    tag: String,
-    @ColorRes backgroundColorRes: Int,
-    @ColorRes textColorRes: Int,
-    onClicked: (() -> Unit)? = null,
-) = postfixWithTag(original, tag as CharSequence, backgroundColorRes, textColorRes, onClicked)
+) = postfixWithTag(
+    original = original,
+    tag = getString(tagRes),
+    tagColor = tagColor,
+    ellipsizeConfiguration = ellipsizeConfiguration,
+    onClicked = onClicked
+)
 
 /**
  * Do not forget to set `movementMethod = LinkMovementMethod.getInstance()` on a TextView to make the tag clickable
  */
 fun Context.postfixWithTag(
-    original: Spannable,
+    original: CharSequence = "",
     tag: CharSequence,
-    @ColorRes backgroundColorRes: Int,
-    @ColorRes textColorRes: Int,
+    tagColor: TagColor,
+    ellipsizeConfiguration: SubjectFormatter.EllipsizeConfiguration? = null,
     onClicked: (() -> Unit)? = null,
 ): Spannable {
-    val postfixed = TextUtils.concat(original, Utils.TAG_SEPARATOR + tag)
 
-    return postfixed.toSpannable().apply {
-        val startIndex = original.length + Utils.TAG_SEPARATOR.length
-        val endIndex = startIndex + tag.length
+    fun getConfiguredTag(): CharSequence {
```

Comment:
This should be a String, also make sure no other type is wrong because you often mix up String, CharSequence and Spannable